### PR TITLE
add docker-compose file

### PR DESCRIPTION
**Pull request by coryschwartz**
* fileocin/lotus
  - includes an initialization that pulls form the latest snapshot, similarly to the snap package and DO droplet
  - Runs the lotus daemon by default, which is the most common scenario, and most likely the expected behavior
* filecoin/lotus-all-in-one
  - includes all the lotus utilities you might want
* docker-compose
  - help devs get up and running quckly for local development, just run `docker-compose up -d` and away you go.
  - Has a hint about how to run a miner, but not uncommented by default since many people wouldn't want that.
  - Developers can attach to the fullnode or to the gateway to test against the limited gateway API, test locally before using public gateways.

### CODE REVIEW

File: Dockerfile.lotus
Comment (travisperson) [inline Dockerfile.lotus:85]: `/docker-lotus-entrypoint.sh` doesn't have a help flag

File: scripts/docker-lotus-entrypoint.sh
Comment (coryschwartz) [reply scripts/docker-lotus-entrypoint.sh:21]: Thanks, good point.

File: Dockerfile.lotus
Comment (coryschwartz) [reply Dockerfile.lotus:85]: > `/docker-lotus-entrypoint.sh` doesn't have a help flag

It only does because it exec's to lotus at the end.

Comment (travisperson) [reply Dockerfile.lotus:86]: oh nice

**Comment by BigLep** [comment]
@coryschwartz : per updated Lotus CR policy, can you get an approval on your team, and then merge if maintainers haven't replied within 2 business days?  Ping me on Slack if you need more info here.

**Comment by coryschwartz** [comment]
Thanks @BigLep

@travisperson let me know if you see any problems here, I had this sitting in draft for a while so it's already pretty ancient.

**Review by travisperson — APPROVED** [review]

**Pull request merged**
